backend/src/models/users.go

```diff
@@ -157,4 +157,33 @@ type ResidentProgramClassInfo struct {
 	ProgramID            uint                    `json:"program_id"`
 	PresentAttendance    int                     `json:"present_attendance"`
 	AbsentAttendance     int                     `json:"absent_attendance"`
+	CreditTypes          string                  `json:"credit_types"`
+}
+
+type ResidentProgramClassHistory struct {
+	ProgramName       string      `json:"program_name"`
+	ClassName         string      `json:"class_name"`
+	Status            ClassStatus `json:"status"`
+	DateStatusChanged string      `json:"date_status_changed"`
+	CreditTypes       string      `json:"credit_types"`
+}
+
+type ResidentProgramClassWeeklySchedule struct {
+	ClassName string `json:"class_name"`
+	StartDate string `json:"start_date"`
+	EndDate   string `json:"end_date"`
+	// CreditTypes string `json:"credit_types"`
+	// Status      string `json:"status"`
+}
+
+type Class struct {
+	Name      string     `json:"class_name"`
+	StartDate time.Time  `json:"start_date"`
+	EndDate   *time.Time `json:"end_date"`
+}
+
+type WeeklySchedule struct {
+	Day      string
+	TimeSlot string
+	Classes  []Class
```

Comment:
@corypride We shouldn't need these if we use the existing schedule on class_events.go